tests/containers/podman_quadlet.pm

```diff
@@ -12,34 +12,48 @@ use testapi;
 use serial_terminal qw(select_serial_terminal);
 use version_utils qw(package_version_cmp);
 use Utils::Systemd qw(systemctl);
+use JSON;
 
-sub check_unit_states {
-    my $expected = shift // 'generated';
-    validate_script_output('systemctl --no-pager is-enabled sleeper.service', qr/$expected/, proceed_on_failure => 1);
-    validate_script_output('systemctl --no-pager is-enabled sleeper-volume.service', qr/$expected/, proceed_on_failure => 1);
-}
+my $unit_prefix = '/etc/containers/systemd';
+my $unit_name = 'quadlet-test';
 
-sub run {
-    my ($self, $args) = @_;
+my $build_imagetag = "localhost/nginx";
+my @systemd_build = ("$unit_prefix/$unit_name.build", <<_EOF_);
```

Comment:
While this works, I would say it's better to place files in `data/containers/quadlet` and use the files from there. See e.g. https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/9af8a47087054f20ee37219e9ce68e624b1f5934/lib/containers/container_images.pm#L57 on how to get files from a data folder.

Reply:
I've considered doing that, after @pdostal suggested this to me, but I went with leaving the files in-line as they were before, because of the variable replacements I am performing there. I didn't want to end up performing `sed` replacement operations on the files but if there's a templating engine (like Jinja2 in Python) for Perl, I am inclined to use that instead...

Reply:
In that case I think it's justified to keep them as they are. Agreed.